### ISSUE TITLE
Queued Jobs cannot find `queue` deferred service

### ISSUE THREAD

**Issue by adamgoose** [laravel/framework#7084]
I'm getting the following stack trace on a fresh Laravel 5 install:

```vagrant@homestead:~/Sites/tmp$ artisan queue:work
exception 'ErrorException' with message 'Undefined index: queue' in /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:484
Stack trace:
#0 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(484): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8, 'Undefined index...', '/home/vagrant/S...', 484, Array)
#1 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(187): Illuminate\Foundation\Application->loadDeferredProvider('queue')
#2 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(123): Illuminate\Foundation\Console\Kernel->bootstrap()
#3 [internal function]: Illuminate\Foundation\Console\Kernel->call('inspire')
#4 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Foundation/Console/QueuedJob.php(34): call_user_func_array(Array, Array)
#5 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(126): Illuminate\Foundation\Console\QueuedJob->fire(Object(Illuminate\Queue\Jobs\RedisJob), Array)
#6 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Queue/Jobs/RedisJob.php(47): Illuminate\Queue\Jobs\Job->resolveAndFire(Array)
#7 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(207): Illuminate\Queue\Jobs\RedisJob->fire()
#8 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(159): Illuminate\Queue\Worker->process('redis', Object(Illuminate\Queue\Jobs\RedisJob), 0, 0)
#9 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(107): Illuminate\Queue\Worker->pop(NULL, NULL, 0, 3, 0)
#10 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(67): Illuminate\Queue\Console\WorkCommand->runWorker(NULL, NULL, 0, 128, false)
#11 [internal function]: Illuminate\Queue\Console\WorkCommand->fire()
#12 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Container/Container.php(523): call_user_func_array(Array, Array)
#13 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Console/Command.php(115): Illuminate\Container\Container->call(Array)
#14 /home/vagrant/Sites/tmp/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(253): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Console/Command.php(101): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /home/vagrant/Sites/tmp/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /home/vagrant/Sites/tmp/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /home/vagrant/Sites/tmp/vendor/symfony/console/Symfony/Component/Console/Application.php(126): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /home/vagrant/Sites/tmp/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(91): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /home/vagrant/Sites/tmp/artisan(34): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#21 {main}
vagrant@homestead:~/Sites/tmp$```

This occurs when:

1) I call `Artisan::queue('inspire');` from a controller
2) run `php artisan queue:work`.

Note: This issue will only become apparent after #7083 is merged. To reproduce this issue, if that PR isn't merged, you'll have to alias the contract first.

**Comment by adamgoose** [comment]
> Pushing a closure to the queue works fine.

**Comment by adamgoose** [comment]
Fixed with 3f5863966d90b5f76b3adb3582bc0b33569a3ccc
